addons/sale_purchase/models/sale_order.py

```diff
@@ -17,12 +17,25 @@ def _compute_purchase_order_count(self):
         for order in self:
             order.purchase_order_count = len(order._get_purchase_orders())
 
+    def _set_analytic_account(self):
+        purchase_orders = self._get_purchase_orders()
+        purchase_order_lines = purchase_orders.order_line.filtered(lambda line: line.sale_order_id == self)
+        for order_line in purchase_order_lines:
+            order_line.analytic_distribution = {self.analytic_account_id.id: 100}
+
     def _action_confirm(self):
         result = super(SaleOrder, self)._action_confirm()
         for order in self:
             order.order_line.sudo()._purchase_service_generation()
```

Comment:
Why not using the result of that method to get the POL? 🤔 

Or why not moving your changes inside `_purchase_service_generation` or in the methods called inside that method, for instance: `_purchase_service_prepare_line_values`?